src/sparse_merkle_tree/smt.rs

```diff
@@ -1349,6 +1350,108 @@ impl SparseMerkleTree {
         self.calculate_query_proof_from_result(db, &data)
     }
 
+    fn next_query(query: &QueryProofWithProof, sibling_hash: &[u8]) -> QueryProofWithProof {
+        let d = query.binary_key()[query.height() - 1];
+        let mut next_query = query.clone();
+        if !d {
+            next_query.hash = [query.hash.as_slice(), sibling_hash]
+                .concat()
+                .hash_with_kind(HashKind::Branch);
+        } else {
+            next_query.hash = [sibling_hash, query.hash.as_slice()]
+                .concat()
+                .hash_with_kind(HashKind::Branch);
+        }
+        next_query.slice_bitmap();
+
+        next_query
+    }
+
+    fn is_bitmap_valid(
+        sibling: &QueryProofWithProof,
+        query: &QueryProofWithProof,
+    ) -> Result<bool, SMTError> {
+        let is_sibling_empty = utils::is_empty_hash(&sibling.hash);
+        if (is_sibling_empty && query.binary_bitmap[0])
+            || (!is_sibling_empty && !query.binary_bitmap[0])
+        {
+            return Err(SMTError::InvalidInput(String::from(
+                "bitmap is not consistent with the nodes type",
+            )));
+        }
+        let is_query_empty = utils::is_empty_hash(&query.hash);
+        if (is_query_empty && sibling.binary_bitmap[0])
+            || (!is_query_empty && !sibling.binary_bitmap[0])
+        {
+            return Err(SMTError::InvalidInput(String::from(
+                "bitmap is not consistent with the nodes type",
+            )));
+        }
+        if !utils::array_equal_bool(&query.binary_bitmap[1..], &sibling.binary_bitmap[1..]) {
+            return Err(SMTError::InvalidInput(String::from(
+                "nodes do not share common path",
+            )));
+        }
+
+        Ok(true)
+    }
+
+    fn prepare_result_proof(
+        proof: &Proof,
+        adding_sibling_hashes: &[(usize, Vec<u8>)],
+        removing_sibling_hashes: &NestedVec,
+        removing_keys: &[&[u8]],
+        next_sibling_hash_index: usize,
+    ) -> Result<Proof, SMTError> {
+        let mut new_sibling_hashes = proof.sibling_hashes.clone();
```

Comment:
```suggestion
        let mut sibling_hashes = proof.sibling_hashes.clone();
```

or

```suggestion
        let mut joined_sibling_hashes = proof.sibling_hashes.clone();
```

or maybe some other appropriate name?

Reply:
Done.